### DISCUSSION TITLE
Help me with internatiolization for project

### DISCUSSION THREAD

**Discussion by nicitaacom** [i18next/next-i18next#2223] [Q&A]
### Step 1
git clone - https://github.com/nicitaacom/acc2-i18n

### Step 2
`pnpm i`
`pnpm dev`

![image](https://github.com/i18next/i18next/assets/39565703/7598d5bd-985c-42d5-ad12-0ef3dd3c19dd)

If you know how to male internatiolization using Next + TypeScript -
Please send example
or
Make PR

To make PR
1. Fork this repository
2. Choose 'for contribution option'
3. Make your changes and make it actually work
4. Create PR

Thank you!

**Comment by adrai** [comment]
https://github.com/i18next/next-i18next/tree/master/examples/simple

**Reply by nicitaacom** [reply]
No way to clone this project
![image](https://github.com/i18next/next-i18next/assets/39565703/ef71242f-28d0-442c-89d4-54c71f59f624)

**Reply by nicitaacom** [reply]
![image](https://github.com/i18next/next-i18next/assets/39565703/1177b666-c2f8-4cd7-af2c-6fcd7f8f6ad0)

**Reply by nicitaacom** [reply]
Also you use ISR - only for locales like /de/somePage en/somePage
But I want to use it for /de/user/username
![image](https://github.com/i18next/next-i18next/assets/39565703/a017c3dd-a9cb-4eaf-9035-ccee860ca738)


Also as I now this function names `generateStaticParams()` instead of `getStaticPaths()`

So I ask you about updated example with SSG

**Reply by nicitaacom** [reply]
Also I would Like to use TypeScript version

**Reply by adrai** [reply]
git clone https://github.com/i18next/next-i18next.git
cd next-i18next/examples/simple
npm i
npm run dev

**Comment by adrai** [comment]
btw: it seems you're using next-intl: https://github.com/nicitaacom/acc2-i18n/blob/development/i18n.ts

**Reply by adrai** [reply]
beside that, you're using the newer app router setup... and not the pages setup...
so, you need to compare that with this: https://locize.com/blog/next-13-app-dir-i18n/
Here a TypeScript example: https://github.com/i18next/next-13-app-dir-i18next-example-ts

**Reply by nicitaacom** [reply]
Thank you

I try it out and give you my feedback soon

**Reply by nicitaacom** [reply]
I come back with my feedback (please take it as point to growth - you have good library and may have better one)
1. You use not last next.js version - I made bump to 13.5.7
2. When I want to deploy my app (to vecel for example) I got a bunch of errors when `pnpm build`
![image](https://github.com/i18next/next-i18next/assets/39565703/fbd08500-9581-48a9-8de6-7c15f7f12325)


If you know hot to fix this - please let me know (because eslint-disable-next-line in each file its not fun)
![image](https://github.com/i18next/next-i18next/assets/39565703/03662ce7-c651-492b-a643-64f212afe2b7)

When you fininsh fixing this error I'd like to contribute to improve your docs (because for me as developer who use i18n frist time it looks a bit complicated - I have skills in writing docs for project so I improve readme.md in https://github.com/i18next/next-13-app-dir-i18next-example-ts)

**Comment by nicitaacom** [comment]
Any updates?

**Comment by nicitaacom** [comment]
Still no updates...

Example just throw error on build
I even don't know how to use this

I want to close this because I will use another library for internationalization

**Reply by adrai** [reply]
Have you checked the example repository... in the meantime there have been some updates: https://github.com/i18next/next-app-dir-i18next-example-ts

i.e. the useTranslation hook/function has been renamed etc...
